### PR TITLE
Dev/kduran/us vbs22 35 inicio de sesión

### PR DESCRIPTION
**Pull request by kendallduranv**
Solicitude to merge this to the development branch. User login functinality added, Algoso unit testing for  login and new user fixed and added

### CODE REVIEW

File: demo/src/main/java/com/project/demo/logic/entity/country/Country.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import java.util.Date;
+import java.util.List;
+
+@Table(name = "VO_Country")
+@Entity
+public class Country {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long country_id;
```
Comment (nefi-cenfo) [inline demo/src/main/java/com/project/demo/logic/entity/country/Country.java]: @mur-tz recommends you use int as the type for id, if I'm not mistaken.

File: demo/src/main/java/com/project/demo/logic/entity/country/CountryRepository.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import com.project.demo.logic.entity.user.User;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface CountryRepository extends JpaRepository<Country, Long>  {
+    @Query("SELECT u FROM Country u WHERE LOWER(u.country_name) LIKE %?1%")
+    List<Country> findUsersWithCharacterInName(String character);
```
Comment (nefi-cenfo) [inline demo/src/main/java/com/project/demo/logic/entity/country/CountryRepository.java]: Do you mean findCountries... instead?

File: demo/src/main/java/com/project/demo/logic/entity/rol/AdminSeeder.java [hunk 3]
```diff
@@ -15,44 +16,50 @@ public class AdminSeeder implements ApplicationListener<ContextRefreshedEvent> {
...
-        var user = new User();
-        user.setName(superAdmin.getName());
-        user.setLastname(superAdmin.getLastname());
-        user.setEmail(superAdmin.getEmail());
-        user.setPassword(passwordEncoder.encode(superAdmin.getPassword()));
-        user.setRole(optionalRole.get());
-
-        userRepository.save(user);
-    }
+//    private void createSuperAdministrator() {
```
Comment (nefi-cenfo) [inline demo/src/main/java/com/project/demo/logic/entity/rol/AdminSeeder.java]: If you don't need the commented code, then you can delete it.

File: demo/src/main/java/com/project/demo/logic/entity/rol/Role.java [hunk 4]
```diff
@@ -6,66 +6,99 @@
...
-@Table(name = "rol")
+
 @Entity
+@Table(name = "VO_UserRole")
 public class Role {
     @Id
     @GeneratedValue(strategy = GenerationType.AUTO)
     @Column(nullable = false)
-    private Integer id;
+    private Integer role_id;
```
Comment (nefi-cenfo) [inline demo/src/main/java/com/project/demo/logic/entity/rol/Role.java]: I guess the type should be 'int'.

File: demo/src/main/java/com/project/demo/logic/entity/rol/RoleEnum.java [hunk 5]
```diff
@@ -3,5 +3,5 @@
 public enum RoleEnum {
     USER,
     ADMIN,
-    SUPER_ADMIN
+//    SUPER_ADMIN
```
Comment (nefi-cenfo) [inline demo/src/main/java/com/project/demo/logic/entity/rol/RoleEnum.java]: If you don't need the commented code, then you can delete it.

File: demo/src/main/java/com/project/demo/logic/entity/rol/RoleSeeder.java [hunk 6]
```diff
@@ -23,11 +23,11 @@ public void onApplicationEvent(ContextRefreshedEvent contextRefreshedEvent) {
...
 
     private void loadRoles() {
-        RoleEnum[] roleNames = new RoleEnum[] { RoleEnum.USER, RoleEnum.ADMIN, RoleEnum.SUPER_ADMIN };
+        RoleEnum[] roleNames = new RoleEnum[] { RoleEnum.USER, RoleEnum.ADMIN };
         Map<RoleEnum, String> roleDescriptionMap = Map.of(
                 RoleEnum.USER, "Default user role",
-                RoleEnum.ADMIN, "Administrator role",
-                RoleEnum.SUPER_ADMIN, "Super Administrator role"
+                RoleEnum.ADMIN, "Administrator role"
+//                RoleEnum.SUPER_ADMIN, "Super Administrator role"
```
Comment (nefi-cenfo) [inline demo/src/main/java/com/project/demo/logic/entity/rol/RoleSeeder.java]: If you don't need the commented code, then you can delete it.

File: demo/src/main/java/com/project/demo/logic/entity/user/User.java [hunk 7]
```diff
@@ -11,90 +12,103 @@
...
 
-@Table(name = "\"user\"")
+@Table(name = "VO_User")
 @Entity
 public class User implements UserDetails {
+
     @Id
     @GeneratedValue(strategy = GenerationType.IDENTITY)
-    private Long id;
+    private Long user_id;
```
Comment (nefi-cenfo) [inline demo/src/main/java/com/project/demo/logic/entity/user/User.java]: Please check the type for the id.

File: demo/src/main/java/com/project/demo/logic/entity/user/User.java [hunk 8]
```diff
@@ -11,90 +12,103 @@
...
+    private Integer creation_responsible;
     @UpdateTimestamp
     @Column(name = "updated_at")
-    private Date updatedAt;
+    private Date last_update_datetime;
+
+    @Column(nullable = true)
+    private Integer update_responsible;
+
+//    @OneToMany(mappedBy = "vo_country")
```
Comment (nefi-cenfo) [inline demo/src/main/java/com/project/demo/logic/entity/user/User.java]: If you don't need the commented code, then you can delete it.

**Review by nefi-cenfo — DISMISSED** [review]

**Review by ichivancrCenfotec — APPROVED** [review]

**Review by mur-tz — APPROVED** [review]

**Review by nefi-cenfo — APPROVED** [review]

**Pull request merged**